spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/AsyncCassandraTemplate.java

```diff
@@ -456,11 +470,12 @@ private ListenableFuture<Boolean> doDelete(Query query, Class<?> entityClass, Cq
 				tableName);
 		SimpleStatement delete = builder.build();
 
-		maybeEmitEvent(new BeforeDeleteEvent<>(delete, entityClass, tableName));
+		maybeEmitEvent(() -> new BeforeDeleteEvent<>(delete, entityClass, tableName));
```

Comment:
I'm actually curious if this does make a difference. After all the closure is an object that needs to be created as well, and I assume a `BeforeDeleteEvent` is a pretty simple object just storing a couple of fields as well.

If it does make a difference I'd love to learn, where the difference comes from.

Reply:
Probably going onto a very detailed level here. There are two aspects. Allocation-wise, you are probably right. A supplier creates another object. However, I would expect the JIT to inline the supplier code with the `maybeEmit` method and other called methods so that the JIT optimizes the supplier away.

Some events might require further analysis (such as ID extraction) and that would be guarded by the supplier as well to not run the code if the event publishing is disabled.